vpr/src/server/task.h

```diff
@@ -19,36 +19,162 @@ namespace server {
+
+    /**
+     * @brief Checks if the options of this task match the options of another task.
+     * 
+     * This method compares the options of this task with the options of another task
+     * and returns true if they match, otherwise returns false.
+     * 
+     * @param other The other task to compare options with.
+     * @return True if the options match, false otherwise.
+     */
+    bool options_match(const std::unique_ptr<Task>& other);
+
+    /**
+     * @brief Retrieves the response buffer.
+     * 
+     * This method returns a constant reference to the response buffer, which contains the data after task execution.
+     * 
+     * @return A constant reference to the response buffer.
+     */
     const std::string& response_buffer() const { return m_response_buffer; }
 
+    /**
+     * @brief Checks if the task has finished execution.
+     * 
+     * This method returns true if the task has finished execution; otherwise, it returns false.
+     * 
+     * @return True if the task has finished execution, false otherwise.
+     */
     bool is_finished() const { return m_is_finished; }
+
+    /**
+     * @brief Checks if the task has encountered an error.
+     * 
+     * This method returns true if the task has encountered an error; otherwise, it returns false.
+     * 
+     * @return True if the task has encountered an error, false otherwise.
+     */
     bool has_error() const { return !m_error.empty(); }
+
+    /**
+     * @brief Retrieves the error message associated with the task.
+     * 
+     * This method returns the error message associated with the task, if any.
+     * 
+     * @return A constant reference to the error message string.
+     */
     const std::string& error() const { return m_error; }
 
+    /**
+     * @brief Retrieves the original number of response bytes.
+     * 
+     * This method returns the original number of response bytes before any chopping operation.
+     * 
+     * @return The original number of response bytes.
+     */
     std::size_t orig_reponse_bytes_num() const { return m_orig_reponse_bytes_num; }
 
+    /**
+     * @brief Checks if the response has been fully sent.
+     * 
+     * This method returns true if the entire response has been successfully sent, 
+     * otherwise it returns false.
+     * 
+     * @return True if the response has been fully sent, false otherwise.
+     */
     bool is_response_fully_sent() const { return m_is_response_fully_sent; }
 
+    /**
+     * @brief Marks the task as failed with the specified error message.
+     * 
+     * This method sets the task's error message to the provided error string, 
+     * indicating that the task has failed.
+     * 
+     * @param error The error message describing the reason for the task's failure.
+     */
     void fail(const std::string& error);
+
+    /**
+     * @brief Marks the task as successfully completed.
+     */
     void success();
+
```

Comment:
I think a better name for this function would be set_success().

Reply:
solved 86d31a38dd75b9464239b8cbce306ea583071565